examples/sampling/graphbolt/lightning/node_classification.py

```diff
@@ -69,10 +70,32 @@ def forward(self, blocks, x):
                 h = self.dropout(h)
         return h
 
+    def log_node_and_edge_counts(self, blocks):
+        node_counts = [block.num_src_nodes() for block in blocks] + [
+            blocks[-1].num_dst_nodes()
+        ]
+        edge_counts = [block.num_edges() for block in blocks]
+        for i, c in enumerate(node_counts):
+            self.log(
+                f"num_nodes/{i}",
+                float(c),
+                prog_bar=True,
+                on_step=True,
+                on_epoch=False,
+            )
+            if i < len(edge_counts):
+                self.log(
+                    f"num_edges/{i}",
+                    float(edge_counts[i]),
+                    prog_bar=True,
+                    on_step=True,
+                    on_epoch=False,
+                )
+
     def training_step(self, batch, batch_idx):
         # TODO: Move this to the data pipeline as a stage.
         blocks = [block.to("cuda") for block in batch.to_dgl_blocks()]
-        x = blocks[0].srcdata["feat"]
+        x = blocks[0].srcdata["feat"].float()
```

Comment:
shall we change the data type to float or better to update the model's data type to accommodate the input?

Reply:
From what I remember, the products dataset didn't have the double datatype. Do you know why the dataset is stored in double?